client/api/factory.go

```diff
@@ -0,0 +1,37 @@
+package api
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/lidofinance/dc4bc/client/api/http_api"
+	"github.com/lidofinance/dc4bc/client/config"
+	"os"
+)
+
+type IServerAbstractFactory interface {
```

Comment:
Interface names do not start with an I, it's very Java style
https://golang.org/doc/effective_go#interface-names